From 7ed26945de2cb42a44d1bc0d586308d36e765f35 Mon Sep 17 00:00:00 2001
From: Venkataramana Kotakonda <venkataramana.kotakonda@intel.com>
Date: Wed, 7 Oct 2015 21:26:12 +0530
Subject: [PATCH] Wcove-gpio: Always supply vconn to fusb

Vconn supply to fusb should be always present and based on current
device state, the vconn supply to port partner will be controlled
by fusb driver.

Change-Id: Ib7027eec9d735e9f59e8e94aa851a11484316c12
Signed-off-by: Venkataramana Kotakonda <venkataramana.kotakonda@intel.com>
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-7236
---
 drivers/misc/intel_wcove_gpio.c | 7 +++----
 1 file changed, 3 insertions(+), 4 deletions(-)

diff --git a/drivers/misc/intel_wcove_gpio.c b/drivers/misc/intel_wcove_gpio.c
index e30b3cb..6b1be4f 100644
--- a/drivers/misc/intel_wcove_gpio.c
+++ b/drivers/misc/intel_wcove_gpio.c
@@ -162,10 +162,6 @@ static void wcgpio_ctrl_worker(struct work_struct *work)
 		gpiod_set_value_cansleep(info->gpio_otg,
 						evt->is_src_connected);
 
-		/* enable/disable vconn based on the provider(source) event */
-		gpiod_set_value_cansleep(info->gpio_vconn,
-						evt->is_src_connected);
-
 		/* FIXME: vchrgin GPIO is not setting here to select
 		 * Wireless Charging */
 		list_del(&evt->node);
@@ -306,6 +302,9 @@ static int wcove_gpio_probe(struct platform_device *pdev)
 
 	check_initial_events(info);
 
+	/* Enable vconn always to typec chip */
+	gpiod_set_value_cansleep(info->gpio_vconn, 1);
+
 	return 0;
 
 error_gpio:
-- 
1.9.1

